app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt

```diff
@@ -144,11 +150,26 @@ class NotificationFragment : Fragment() {
         }
     }
 
+    private val refreshNotifications = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            customNotificationList.add(Notification(intent.getStringExtra("text")!!, intent.getStringExtra("createdat")!!))
```

Comment:
Use string constants, define in this fragment itself in companion object. Also avoid !! null check